crates/wasmcloud-host/src/messagebus/handlers.rs

```diff
@@ -439,7 +439,7 @@ impl Handler<QueryProviders> for MessageBus {
                 .subscribers
                 .keys()
                 .filter_map(|k| match k {
-                    WasmCloudEntity::Capability { id, .. } => Some(id.to_string()),
+                    c @ WasmCloudEntity::Capability { .. } => Some(c.clone()),
```

Comment:
What is this wizardry? you can assign the matched struct to a variable instead of dereferencing fields?

Reply:
Yes. The variable will still be of the enum type and not the variant so it only comes in handy in a few places.